lib/model.js

```diff
@@ -1000,16 +1000,18 @@ Connection.from = function(url) {
  * Helper function to improve the Atlas user experience by
  * providing better default values.
  *
- * @param {String} url  The connection string URL.
- * @param {String} mongodb_password  The
+ * @param {String} url - The connection string URL.
+ * @param {String} mongodb_password - The mongodb_password
+ *   which the user may need to change.
  * @returns {Object} Connection attributes to override
  * @private
  */
 Connection._improveAtlasDefaults = function(url, mongodb_password) {
   var atlasConnectionAttrs = {};
   if (Connection.isAtlas(url)) {
-    atlasConnectionAttrs.ssl = 'UNVALIDATED';
-    if (mongodb_password === 'PASSWORD') {
+    atlasConnectionAttrs.ssl = 'SYSTEMCA';
+    if (mongodb_password.length <= 20 &&
```

Comment:
to simplify and not introduce magic number, use a test like `.match(/.?PASSWORD.?/i)` which should test for PASSWORD and potentially one character on either side of the string.